CRM/Activity/BAO/Activity.php

```diff
@@ -1398,9 +1398,6 @@ public static function sendSMSMessage(
 
     $providerObj = CRM_SMS_Provider::singleton(['provider_id' => $smsProviderParams['provider_id']]);
     $sendResult = $providerObj->send($recipient, $smsProviderParams, $tokenText, NULL, $sourceContactID);
-    if (PEAR::isError($sendResult)) {
-      throw new CRM_Core_Exception($sendResult->getMessage());
-    }
```

Comment:
I guess I can test this but I think most SMS extensions specifically call PEAR::raiseError, so does that internally now get converted automatically to an exception or would that still be a pear error?

Reply:
It would get converted because this is changing the PEAR error callback https://github.com/civicrm/civicrm-core/pull/19323/files#diff-dc1c0506257784ef60b0b3f296a5e64c633c0a19d47dbd87885eb21852836fd6R89 to be this [function](https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/Error.php#L940) instead of this [one](https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/Error.php#L138)

Reply:
So no more familiar yellow screen? :cry:  I'll get over it...

Tested and yes it errors with a PEAR_Exception now.

Reply:
It will because if Exception isn't handled it should wind up here https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/Error.php#L438

Reply:
Oh I see, you just have to scroll down. 😀